ui/src/containers/ConfigScreen/index.tsx

```diff
@@ -57,102 +75,381 @@ const ConfigScreen: React.FC = function () {
+    validateOtherKeysByApi: boolean
+  ) => {
+    const checkMultiConfigKeys = (multiConfigKeys: any) => {
+      const invalidKeys: any = {};
+
+      Object.entries(multiConfigKeys || {}).forEach(([configKey, config]) => {
+        Object.entries(config || {}).forEach(([key, value]) => {
+          if (typeof value === "string" && value.trim() === "") {
+            if (!invalidKeys?.[configKey]) {
+              invalidKeys[configKey] = [];
+            }
+            invalidKeys?.[configKey].push(key);
+          }
+        });
+      });
 
-    const iv = CryptoJS?.lib?.WordArray?.random(128 / 8);
+      return invalidKeys;
+    };
 
-    const encrypted = CryptoJS?.AES?.encrypt(msg, key, {
-      iv,
-      padding: CryptoJS?.pad?.Pkcs7,
-      mode: CryptoJS?.mode?.CBC,
-    });
+    const checkOtherKeys = (keys: any) => {
+      const invalidKeys: string[] = [];
 
-    // salt, iv will be hex 32 in length
-    // append them to the ciphertext for use  in decryption
-    const transitmessage =
-      (salt?.toString() ?? "") +
-      (iv?.toString() ?? "") +
-      (encrypted?.toString() ?? "");
-    return transitmessage;
-  };
+      Object.entries(keys || {}).forEach(([key, value]) => {
+        if (typeof value === "string" && value.trim() === "") {
+          invalidKeys.push(key);
+        }
+      });
 
-  const decryptData = (transitmessage: any, pass: any) => {
-    const salt = CryptoJS?.enc?.Hex?.parse(transitmessage?.substr(0, 32));
-    const iv = CryptoJS?.enc?.Hex?.parse(transitmessage?.substr(32, 32));
-    const encrypted = transitmessage?.substring(64);
+      return invalidKeys;
+    };
 
-    const key = CryptoJS?.PBKDF2(pass, salt, {
-      keySize: keySize / 32,
-      iterations,
-    });
+    let normalInvalidKeys = {};
+    let serverNormalInvalidKeys = {};
 
-    const decrypted = CryptoJS?.AES?.decrypt(encrypted, key, {
-      iv,
-      padding: CryptoJS?.pad?.Pkcs7,
-      mode: CryptoJS?.mode?.CBC,
-    });
-    return decrypted?.toString(CryptoJS?.enc?.Utf8);
+    if (!validateMultiConfigKeysByApi) {
+      normalInvalidKeys = checkMultiConfigKeys(
+        configurationData?.multi_config_keys
+      );
+      serverNormalInvalidKeys = checkMultiConfigKeys(
+        serverConfiguration?.multi_config_keys
+      );
+    }
+
+    const normalInvalidKeysList = [
+      ...Object.entries(normalInvalidKeys)
+        .filter(([keys]) => keys?.length)
+        .map(([source, keys]) => ({ source, keys })),
+      ...Object.entries(serverNormalInvalidKeys)
+        .filter(([keys]) => keys?.length)
+        .map(([source, keys]) => ({ source, keys })),
+    ];
+
+    let otherInvalidKeysList: { source: string; keys: any[] }[] = [];
+    if (!validateOtherKeysByApi) {
+      const otherInvalidKeys = checkOtherKeys(configurationData);
+      const serverOtherInvalidKeys = checkOtherKeys(serverConfiguration);
+
+      otherInvalidKeysList = [
+        ...otherInvalidKeys.map((key) => ({
```

Comment:
otherInvalidKeys?.

Reply:
After adding optional chaining getting an Unsafe usage of optional chaining. If it short-circuits with 'undefined' the evaluation will throw TypeError